crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -20,25 +26,41 @@ pub use {auction::Auction, solution::Solution, transaction::Transaction};
 #[derive(Debug)]
 pub struct Settlement {
     solution: Solution,
+    transaction: Transaction,
     auction: Auction,
 }
 
 impl Settlement {
     pub async fn new(
-        tx: &Transaction,
+        transaction: Transaction,
         domain_separator: &eth::DomainSeparator,
         persistence: &infra::Persistence,
     ) -> Result<Self, Error> {
-        let solution = Solution::new(&tx.input, domain_separator)?;
+        let solution = Solution::new(&transaction.input, domain_separator)?;
         let auction = persistence.get_auction(solution.auction_id()).await?;
 
-        Ok(Self { solution, auction })
+        Ok(Self {
+            solution,
+            transaction,
+            auction,
+        })
     }
 
     /// CIP38 score calculation
     pub fn score(&self) -> Result<competition::Score, solution::error::Score> {
         self.solution.score(&self.auction)
     }
+
+    /// Returns the observation of the settlement.
+    pub fn observation(&self) -> Observation {
```

Comment:
Is this the object that is meant to replace `SettlementUpdate`? I think we should simply pass the `Settlement` object and make all those fields getter variables on `Settlement`